idaes/models_extra/co2_capture_and_utilization/unit_models/membrane_1d.py

```diff
@@ -0,0 +1,321 @@
+        assumes that temperature remains constant on each side of the membrane.
+
+        """
+        super().build()
+
+        feed_dict = dict(self.config.feed_side)
+        sweep_dict = dict(self.config.sweep_side)
+
+        feed_dict["flow_direction"] = FlowDirection.forward
+        if self.config.flow_type == MembraneFlowPattern.COCURRENT:
+            sweep_dict["flow_direction"] = FlowDirection.forward
+        elif self.config.flow_type == MembraneFlowPattern.COUNTERCURRENT:
+            sweep_dict["flow_direction"] = FlowDirection.backward
+        else:
+            raise ConfigurationError(
+                f"{self.name} Membrane1D only supports cocurrent and "
+                "countercurrent flow patterns, but flow_type configuration"
+                " argument was set to {config.flow_type}."
+            )
+
+        if self.config.sweep_flow is False:
+            sweep_dict["has_feed"] = False
+
+        streams_dict = {"feed_side": feed_dict, "sweep_side": sweep_dict}
+        self.mscontactor = MSContactor(
+            streams=streams_dict,
+            number_of_finite_elements=self.config.finite_elements,
+        )
+
+        self.feed_side_inlet = Port(extends=self.mscontactor.feed_side_inlet)
+        self.feed_side_outlet = Port(extends=self.mscontactor.feed_side_outlet)
+        if self.config.sweep_flow is True:
+            self.sweep_side_inlet = Port(extends=self.mscontactor.sweep_side_inlet)
+        self.sweep_side_outlet = Port(extends=self.mscontactor.sweep_side_outlet)
+
+        self._make_geometry()
+        self._make_performance()
+
+    def _make_geometry(self):
+
+        self.area = Var(
+            initialize=100, units=units.cm**2, doc="Area per cell (or finite element)"
+        )
+
+        self.length = Var(initialize=100, units=units.cm, doc="The membrane length")
+        self.cell_length = Expression(expr=self.length / self.config.finite_elements)
+
+        self.cell_area = Var(initialize=100, units=units.cm**2, doc="The membrane area")
+
+        @self.Constraint()
+        def area_per_cell(self):
+            return self.cell_area == self.area / self.config.finite_elements
+
+    def _make_performance(self):
+        feed_side_units = (
+            self.config.feed_side.property_package.get_metadata().derived_units
+        )
+
+        self.permeance = Var(
+            self.flowsheet().time,
+            self.mscontactor.elements,
+            self.mscontactor.feed_side.component_list,
+            initialize=1,
+            doc="Values in Gas Permeance Unit(GPU)",
+            units=units.dimensionless,
+        )
+
+        self.gpu_factor = Param(
+            default=10e-8 / 13333.2239,
+            units=units.m / units.s / units.Pa,
+            mutable=True,
+            doc=" This is a coefficient that will convert the unit of permeability from GPU to SI units for further calculation",
+        )
+
+        """
```

Comment:
Pylint complains about this - general practice is to do this as single line comments (i.e. use `#`).